src/gyro/gyroCk.ts

```diff
@@ -0,0 +1,96 @@
+const GyroMsg_ck = `
+public class GyroMsg {
+    float gyroX;
+    float gyroY;
+    float gyroZ;
+
+    function float getGyroX() {
+        return gyroX;
+    }
+
+    function float getGyroY() {
+        return gyroY;
+    }
+
+    function float getGyroZ() {
+        return gyroZ;
+    }
+
+    function void _copy(GyroMsg localMsg) {
+        localMsg.gyroX => gyroX;
+        localMsg.gyroY => gyroY;
+        localMsg.gyroZ => gyroZ;
+    }
+}
+`;
+
+const Gyro_ck = `
+global Event _gyroReading;
+global int _gyroActive;
+
+global float _gyroX;
+global float _gyroY;
+global float _gyroZ;
+
+public class Gyro extends Event {
+
+    0 => int isGyroOpen;
+    0 => int active;
+
+    string deviceName; 
+
+    // GyroMsg Queue
+    GyroMsg _gyroMsgQueue[0];
+
+    function string name() {
+        return deviceName;
+    }
+
+    function int openGyro(int num) {
+        if (num < 0) {
+            false => active;
+        } else {
+            "js DeviceOrientationEvent" => deviceName;
```

Comment:
nit: Are we calling the Gyro device "js DeviceOrientationEvent" or is it like `js DeviceOrientationSensor`?